kernel/task/src/lib.rs

```diff
@@ -1181,88 +1159,152 @@ impl Drop for JoinableTaskRef {
 }
 
 
-/// A shareable, cloneable reference to a `Task` that exposes more methods
-/// for task management and auto-derefs into an immutable `&Task` reference.
-/// 
-/// The `TaskRef` type is necessary because in many places across Theseus,
-/// a reference to a Task is used. 
-/// For example, task lists, task spawning, task management, scheduling, etc. 
-/// 
-/// ## Equality comparisons
-/// `TaskRef` implements the [`PartialEq`] and [`Eq`] traits to ensure that
-/// two `TaskRef`s are considered equal if they point to the same underlying `Task`.
-#[derive(Debug, Clone)]
-pub struct TaskRef(Arc<Task>);
-
-impl TaskRef {
-    /// Creates a new `TaskRef`, a shareable wrapper around the given `Task`.
-    /// 
-    /// This does *not* add this task to the system-wide task list or any runqueues,
-    /// nor does it schedule this task in.
-    /// 
-    /// ## Return
-    /// Returns a [`JoinableTaskRef`], which derefs into the newly-created `TaskRef`
-    /// and can be used to "join" this task (wait for it to exit) and obtain its exit value.
-    pub fn new(task: Task) -> JoinableTaskRef {
-        let taskref = TaskRef(Arc::new(task));
-
-        // Mark this task as joinable, now that it has been wrapped in the proper type.
-        taskref.joinable.store(true, Ordering::Relaxed);
-        JoinableTaskRef { task: taskref }
+/// A wrapper around `TaskRef` that allows this task to mark itself as exited.
+///
+/// This is only obtainable when a task is first switched to, specifically while
+/// it is executing the `spawn::task_wrapper()` function
+/// (before it proceeds to running its actual entry function).
+///
+/// ## Not `Clone`-able
```

Comment:
I think `ExitableTaskRef` can implement `Clone` and still be correct.

Reply:
sure, that may be true. As I said in the other convo, it's an internal type so it doesn't really matter. 

I guess the question is under what scenario would we need or want to clone an `ExitableTaskRef`?